processing/src/main/java/org/apache/carbondata/processing/loading/sort/unsafe/UnsafeSortDataRows.java

```diff
@@ -188,13 +197,19 @@ public void addRowBatchWithOutSync(Object[][] rowBatch, int size)
   }
 
   private void addBatch(Object[][] rowBatch, int size) throws CarbonSortKeyAndGroupByException {
+    if (rowPage == null) {
```

Comment:
No need to check for null here as this scenario will never occur

Reply:
when other iterator enters , it will be made null by previous iterator in case of failure. hence it is required.